integration/docker-compose.yml

```diff
@@ -0,0 +1,28 @@
+version: '3.7'
```

Comment:
For what purpose do we need docker in the integration tests? Can't we just do `ng serve` or `npm start`?

Reply:
Both applications (shop and portal) are built in the production mode and webpack emit their assets to `dist` folders. We would like to have some kind of web servers that would serve those static files and listen appropriate ports. Docker is super helpful here since it requires a little bit of configuration and it's able to run both applications using just 1 command.

Might be harder to accomplish the same behaviour using `npm start` or other commands.

Reply:
I would rather not require contributors and the CI to install docker just to run the tests. You can run multiple servers in a single command using [concurrently](https://www.npmjs.com/package/concurrently) or similar.

Docker is a great tool, but I think here it adds barriers to contribution, instead of removing them.

```json
{
  "scripts": {
    "start:integration": "concurrently -n w: yarn:start:integration:*",
    "start:integration:portal": "yarn --cwd ...",
    "start:integration:shop": "yarn --cwd ..."
  }
}
```

Reply:
Honestly saying I was thinking about `concurrently`. But it would require to have node.js servers in both apps to serve statics, doesn't it?

> the CI to install docker

Github actions or Circle have pre-installed docker, also Travis provides it via service (requires 1-line command).

Reply:
> I would rather not require contributors

Agreed. In my opinion, we could have both. That Docker image, that is used in the config, is a Go image with 5 MB size (super lightweight and super fast).

And also provide a way to serve statics via another approach for other people who would want to contribute.

Personally, I work mostly all-time with Docker, but it's only a subjective vision for sure :smile:

Reply:
Gonna have to insist here - I use docker as well but think that requiring docker to run tests is a barrier to contribution. This library is already very complex (browser lib, schematics, builder, supporting different versions of angular, etc).

Running two web servers is completely fine.

Reply:
Okay, I will add concurrently then and 2 simple express servers.

Reply:
You can use [serve](https://www.npmjs.com/package/serve) or [http-server](https://www.npmjs.com/package/http-server) to avoid writing your own static file servers. Also, you can use webpack-dev-server to host the files too (which is what I normally do).

Reply:
Serve even has support for SPA mode, where all URLs default to returning the index.html. You can do that with `serve -s`

Reply:
Even better :)

Reply:
Removed docker and added "serve".